### PR TITLE
Add schemas for themebuilder and swatches

### PR DESCRIPTION
**Pull request by joneff**
Needed for https://github.com/telerik/sass-theme-builder/issues/160.

The component names are currently canonical e.g. dataviz, instead of the individual charts. I guess we can change that, if we see fit.

### CODE REVIEW

**Comment by joneff** [comment]
@rkirilov @grozevg will the above schemas work for you?

File: lib/schemas/CssColor.json [hunk 1]
```diff
@@ -0,0 +1,221 @@
+{
+    "$schema": "http://json-schema.org/draft-07/schema",
+    "$id": "kendo-cli://schemas/CssColor.json",
```
Comment (tsvetomir) [inline lib/schemas/CssColor.json]: Shouldn't that be a valid URL?

File: lib/schemas/CssColor.json [hunk 2]
```diff
@@ -0,0 +1,221 @@
+{
+    "$schema": "http://json-schema.org/draft-07/schema",
```
Comment (tsvetomir) [inline lib/schemas/CssColor.json]:
```suggestion
    "$schema": "https://json-schema.org/draft/2020-12/schema",
```

**Review by tsvetomir — COMMENTED** [review]
I'd switch to kebab-case for the schema file names.

**Comment by rkirilov** [comment]
> @rkirilov @grozevg will the above schemas work for you?

I think in the future when we want to distribute partially kendo theme (for particular components only) these schemas will be good solution

File: lib/schemas/CssColor.json (hunk 1 above)
Comment (joneff) [reply lib/schemas/CssColor.json]: yep, it should be. However, I don't know what the real URL should be, so I am leaving it with a pseudo URL fragment that we are to replace.

Slight clarification: while `$id` should be URI, it's not required to be network addressable.

File: lib/schemas/CssColor.json (hunk 2 above)
Comment (joneff) [reply lib/schemas/CssColor.json]: I could change that, but I also need to change `definitions` to `$def`

**Review by eeasss — APPROVED** [review]

File: lib/schemas/Sass.json [hunk 3]
```diff
@@ -0,0 +1,19 @@
+{
+    "$schema": "http://json-schema.org/draft-07/schema",
+    "$id": "kendo-cli://schemas/Sass.json",
+    "title": "Sass schemas",
+
+    "definitions": {
```
Comment (eeasss) [inline lib/schemas/Sass.json]: Is this schema file expected to grow in the future with more descriptors? We link in the swatches file but we could also define the variable there.

File: lib/schemas/CssColor.json [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+            "description": "RGB color",
+            "type": "string",
+            "pattern": "^rgb\\(\\s*\\d{1,3}\\s*,\\s*\\d{1,3}\\s*,\\s*\\d{1,3}\\s*\\)$",
+            "example": "rgb( 255, 255, 255 )"
+        },
+
+        "RgbaColor": {
+            "description": "RGBA color",
+            "type": "string",
+            "pattern": "^rgba\\(\\s*\\d{1,3}\\s*,\\s*\\d{1,3}\\s*,\\s*\\d{1,3}\\s*\\,\\s*(?:0|1|1.0+|0\\.\\d+|\\.\\d+)s*\\)$",
```
Comment (PreslavKozovski) [inline lib/schemas/CssColor.json]:
```suggestion
            "pattern": "^rgba\\(\\s*\\d{1,3}\\s*,\\s*\\d{1,3}\\s*,\\s*\\d{1,3}\\s*\\,\\s*(?:0|1|1.0+|0\\.\\d+|\\.\\d+)\\s*\\)$",
```

File: lib/schemas/CssColor.json [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+            "example": "rgb( 255, 255, 255, 0 )"
```
Comment (PreslavKozovski) [inline lib/schemas/CssColor.json]:
```suggestion
            "example": "rgba( 255, 255, 255, 0 )"
```

File: lib/schemas/Sass.json [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+
+        "variableType": {
+            "description": "Valid Sass variable type",
+            "markdownDescription": "Valid Sass variable type. See [Sass values](https://sass-lang.com/documentation/values)",
+            "type": "string",
+            "enum": [
+                "color",
+                "boolean"
```
Comment (PreslavKozovski) [inline lib/schemas/Sass.json]: Why did you add only these two types? How about `number`, `string`, etc?

For example, in the current ThemeBuilder - `Border radius` has `variableType` - `number`.

File: lib/schemas/Sass.json (hunk 3 above)
Comment (joneff) [reply lib/schemas/Sass.json]: yes, we could indeed define the valid variables, as well as the variable value format i.e. `$*-bg` must accept color, `$*-gradient` must accept gradient, `$enable-*` must be boolean etc.

File: lib/schemas/ThemebuilderSwatch.json [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+        "themeBuilder": {
+            "description": "List of variables to be customized in ThemeBuilder, organized in groups",
+            "type": "array",
+            "items": {
+                "$ref": "#/definitions/group"
+            }
+        }
+    },
+    "additionalProperties": false,
+    "required": [ "$schema", "name", "base", "components", "themeBuilder" ],
```
Comment (PreslavKozovski) [inline lib/schemas/ThemebuilderSwatch.json]:
```suggestion
        },
        "additionalProperties": false,
        "required": [ "$schema", "name", "base", "components", "themeBuilder" ],
    },
```

File: lib/schemas/ThemebuilderSwatch.json [hunk 8]
```diff
@@ -0,0 +1,138 @@
...
+            "properties": {
+                "name": {
+                    "description": "Friendly name for variable. Will be shown in Themebuilder as label for editor",
+                    "type": "string"
+                },
+                "type": {
+                    "$ref": "Sass.json#/definitions/variableType"
+                },
+                "value": {
+                    "type": [ "string", "boolean", "null" ]
```
Comment (PreslavKozovski) [inline lib/schemas/ThemebuilderSwatch.json]:
```suggestion
                    "type": [ "string", "boolean", "null", "number" ]
```

Comment (PreslavKozovski) [reply lib/schemas/ThemebuilderSwatch.json]: We need `number` for valid cases like:

`$my-border-rad: 0;`

File: lib/schemas/ThemebuilderSwatch.json [hunk 9]
```diff
@@ -0,0 +1,138 @@
...
+                }
+            },
+            "allOf": [
```
Comment (PreslavKozovski) [inline lib/schemas/ThemebuilderSwatch.json]: If we decide to add the `number`, we need an additional rule here.

**Review by PreslavKozovski — CHANGES_REQUESTED** [review]

**Review by PreslavKozovski — APPROVED** [review]
